src/CollapseEmbeds/hider.tsx

```diff
@@ -15,11 +15,71 @@ export interface HiderProps {
     placeholders: string[];
     type: AccessoryType;
     children: React.ReactNode;
+    id?: string;
 }
 
-export const Hider = ({placeholders, type, children}: HiderProps): JSX.Element => {
-    const [shown, setShown] = React.useState(!Settings.current.hideByDefault);
-    Settings.useListener(({hideByDefault}) => setShown(!hideByDefault), []);
+export const Hider = ({placeholders, type, children, id}: HiderProps): JSX.Element => {
+    // Update lastSeen timestamp on mount and track component mounted state
+    const mounted = React.useRef(true);
```

Comment:
What is the purpose of this ref? The click handler & settings listener shouldnt run after the component is unmounted anyway.